File: HAL/AIDLClient/app/src/main/java/com/example/aidlclient/MainActivity.java
package com.example.aidlclient;

import android.content.ComponentName;
import android.content.Intent;
import android.content.ServiceConnection;
import android.os.Bundle;
import android.os.IBinder;
import android.os.RemoteException;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.aidliti.IHamada;

public class MainActivity extends AppCompatActivity {

    private IHamada hamada ;

    ServiceConnection service = new ServiceConnection() {
        @Override
        public void onServiceConnected(ComponentName componentName, IBinder iBinder) {
            hamada = IHamada.Stub.asInterface(iBinder);
            TextView t = findViewById(R.id.textview);
            try {
                t.setText(hamada.whatIsYourName());
            } catch (RemoteException e) {
                throw new RuntimeException(e);
            }
        }

        @Override
        public void onServiceDisconnected(ComponentName componentName) {

        }
    };



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        // action of service
        Intent intent = new Intent("com.example.aidliti.Hamada");

        // package
        intent.setPackage("com.example.aidliti");

        // binding this service
        boolean bind = bindService(intent, service, BIND_AUTO_CREATE);




    }
}

File: Android_Development/05_Android_Kotlin/03_Session/03_Session/app/src/main/java/com/example/a03_session/MainActivity.kt
package com.example.a03_session

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var btnStart: Button
    private lateinit var btnStop: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnStart = findViewById(R.id.btn_start)
        btnStop = findViewById(R.id.btn_stop)

        btnStart.setOnClickListener {
            val intent = Intent(this, MystartedService::class.java)
            startService(intent)
        }

        btnStop.setOnClickListener {
            val intent = Intent(this, MystartedService::class.java)
            stopService(intent)
        }
    }
}
